### ISSUE TITLE
If validation set contains multiple files, ground truth is length of longest

### ISSUE THREAD

**Issue by GuyPaddock** [mozilla/TTS#681]
It seems as though the length of the ground truth is influenced by the length of the longest validation audio file, even if there are some audio files in the validation set that are shorter. In my testing, this seemed to cause a studder at the end of each prediction.

- Is that behavior expected?
- Should the validation set be all equal lengths? (My assumption is no).

## Branch
I'm using `dev`, checked out at `547bfc4ce99e7782f98d0403429d15ec23cdc91d`.

## Background Information
My current data set is very small -- only about 158 files or so -- which means I'm not expecting great results. I'm doing transfer learning based on the LJSpeech Tacotron data set, though, to try to get what I can out of the data I have at the moment.

While tinkering, I noticed that if I didn't provide a `meta_file_val`, only a single audio file out of the set of 158 was chosen by the training process to be used for validation. Looking at the following utility method, which generates the validation set from the training set if one is not provided, that makes sense because 10% of the data set is used for validation if a validation set is not provided, and 10% of 158 is 1.58 which rounds down to 1:
https://github.com/mozilla/TTS/blob/d320ccf6c66671087ee69561b651f7f877a584af/TTS/tts/utils/generic_utils.py#L10

I also noticed that when I created a validation set by moving a few of my training set audios to the validation set and ran training, the average error was higher throughout training (95% to 85% after 2,500 epochs) but also the ground truth and predictions for validation had a lot of space at the end:
![image](https://user-images.githubusercontent.com/2631799/110651373-62061300-8189-11eb-9e06-21e3fa63707f.png)

My files have no silence before or after the audio. I've double-checked and there is no silence in any of the validation set files.
The lengths of the files are:
- 2 seconds
- 4 seconds
- 1 seconds
- 1 seconds
- 3 seconds
- 3 seconds
- 2 seconds
